nifi-nar-bundles/nifi-slack-bundle/nifi-slack-services-rtm/src/main/java/org/apache/nifi/processors/slack/controllers/RTMSlackConnectionService.java

```
@@ -0,0 +1,110 @@
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+
+import javax.websocket.DeploymentException;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.lifecycle.OnDisabled;
+import org.apache.nifi.annotation.lifecycle.OnEnabled;
+import org.apache.nifi.annotation.lifecycle.OnShutdown;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.processor.Processor;
+import org.apache.nifi.processor.util.StandardValidators;
+@CapabilityDescription("Implementation of SlackConnectionService." +
+  "This service connects to the Real Time Messaging API of slack.")
+public class RTMSlackConnectionService extends AbstractControllerService implements SlackConnectionService {
+
+  private static final PropertyDescriptor API_TOKEN = new PropertyDescriptor.Builder()
+    .name("api-token")
+    .displayName("API token")
+    .description("Slack auth token required for Real Time Messaging API")
+    .sensitive(true)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .build();
+
+  private ConcurrentHashMap<String, Consumer<String>> map = new ConcurrentHashMap<>();
+
+  private RTMClient rtmClient;
+
+  @OnEnabled
+  public void startClient(ConfigurationContext context) {
+    try {
+      rtmClient = getRtmClient(context);
+      rtmClient.addMessageHandler(this::sendMessage);
+      rtmClient.addCloseHandler(closeReason -> getLogger().info("Slack RTM Client closed: " + closeReason.toString()));
+      rtmClient.addErrorHandler(throwable -> getLogger().error("Slack RTM Client error:", throwable));
+      rtmClient.connect();
+    } catch (IOException|DeploymentException e) {
+      getLogger().error("Error while starting RTM client", e);
```

We should throw these Exceptions so that NiFi flow engine know the CS didn't start correctly. And call this method periodically.